### ISSUE TITLE
Warning when a local version on a GH repo is different from remote

### ISSUE THREAD

**Issue by skptic** [nextflow-io/nextflow#376]
When running NF with 

    nextflow run <organisation>/<repo>

if a local version of a repo exists, yet the it is different from the remote, it would be better if a warning message is provided.

Currently it is not clear to the user that an old local or new remote version is being executed.

Somewhat related to https://github.com/nextflow-io/nextflow/issues/233

Maybe an error message would solve both issues.

**Comment by pditommaso** [comment]
Not sure about this. We should make a remote call each time a run is executed. In alternative it could be showed this information in the `nextflow info <organisation>/<repo>` command.

**Comment by ewels** [comment]
I think this has been discussed quite a few times now? It's a pretty common issue that we run into as well. How bad is it to do a remote call each time?

I guess an alternative could be to print a log message explicitly saying that nextflow is using a cached version of the repo, along with a date and perhaps a diff on the date (eg. "3 weeks ago" type string).

**Comment by pditommaso** [comment]
My understanding is that it's not so trivial. It requires to fetch (or clone) the repository which could be very expensive (depending the repository). 

A lightweight approach could be to use instead the GitHub API to check the last commit (and implement the  same for GitLab, BitBucket).

**Comment by ewels** [comment]
Can we not use `git ls-remote`? This seems to work well for me:

```
$ git ls-remote https://github.com/SciLifeLab/NGI-RNAseq.git HEAD

1a609ef3b68c7f0bac6400c5a2c43a699cb1ec03	HEAD
```

**Comment by ewels** [comment]
(same command works for bitbucket, and I guess other types of remote too)

**Comment by pditommaso** [comment]
Didn't know about that. I'll give a try.

**Comment by ewels** [comment]
Me either actually - just found it on stack overflow. Looks nice thought! 😆

**Comment by pditommaso** [comment]
I've added this message when a different commit id is available remotely: 

```
NOTE: Your local project version looks outdated - a different revision is available in the remote repository [$commitId]
```

Tho, I'm still not super happy. It takes 1-2 second to check it (depending the connection). It could be fair considering a pipeline runs for hours.

I'm a bit more worried because there are use cases in which people want just to use their local version, and this message could be annoying after a while. 

I've also added this information in the `nextflow info <repo>` command. If you want to give a try use `0.27.0-SNAPSHOT` version .
